PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/TunnelProxy.java

```diff
@@ -0,0 +1,100 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.appservice;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.jcraft.jsch.*;
+import com.microsoft.azure.management.appservice.PublishingProfile;
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.azure.toolkit.lib.common.utils.WebSocketSSLProxy;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+public class TunnelProxy {
+    private static final Logger logger = Logger.getLogger(TunnelProxy.class.getName());
+    private static final String LOCALHOST = "localhost";
+    private static final String DEFAULT_SSH_USERNAME = "root";
+    private static final String DEFAULT_SSH_PASSWORD = "Docker!";
+    private WebAppBase webAppBase;
+    private WebSocketSSLProxy wssProxy;
+
+    public TunnelProxy(WebAppBase webAppBase) {
+        Preconditions.checkArgument(Objects.nonNull(webAppBase), "Null `webAppBase` parameter is not allowed for class: TunnelProxy");
+        this.webAppBase = webAppBase;
+        reset();
+    }
+
+    public void reset() {
+        String host = webAppBase.defaultHostName().toLowerCase().replace("http://", "").replace("https://", "");
+        String[] parts = host.split("\\.", 2);
+        host = Joiner.on('.').join(parts[0], "scm", parts[1]);
+        PublishingProfile publishingProfile = webAppBase.getPublishingProfile();
+        wssProxy = new WebSocketSSLProxy(String.format("wss://%s/AppServiceTunnel/Tunnel.ashx", host),
+                                         publishingProfile.gitUsername(), publishingProfile.gitPassword());
+    }
+
+    public void close() {
+        if (wssProxy != null) {
+            wssProxy.close();
+            wssProxy = null;
+        }
+    }
+
+    public String executeCommandViaSSH(String cmd) throws IOException {
+        wssProxy.start();
+        return executeSSHCommand(cmd, wssProxy.getLocalPort());
+    }
+
+    private static String executeSSHCommand(String command, int port) throws IOException {
+        try {
+            Properties config = new Properties();
+            config.put("StrictHostKeyChecking", "no");
+            JSch jsch = new JSch();
+            Session session = jsch.getSession(DEFAULT_SSH_USERNAME, LOCALHOST, port);
+            session.setPassword(DEFAULT_SSH_PASSWORD);
+            session.setConfig(config);
+            session.connect();
+            Channel channel = session.openChannel("exec");
+            ((ChannelExec) channel).setCommand(command);
+            channel.setInputStream(null);
```

Comment:
`setInputStream(null)` but read by `InputStream in = channel.getInputStream()` below?